### PR TITLE
Add ensure! convenience macro

### PR DESCRIPTION
**Pull request by aldanor**
This is quite trivial but is something I found myself defining in most of my crates (because you would always have some sort of kind of if-condition for `bail!` and quite often this condition is a trivial expression that does not affect the error kind/message, particularly so in i/o heavy blocks; less typing, less braces and less rightward-drift = win), so I figured I'd share as I think it fits the spirit of this crate quite well.

Basically,

```rust
ensure!(ultimate_answer == 42, "some {}", "error");
```

### CODE REVIEW

**Comment by Yamakaky** [comment]
Hum, interesting! Could you update the changelog?
Also, I'm thinking about `ensure_eq`. Not sure if it would be interesting, since unlike `assert_eq` there would be no difference in the output.

**Comment by aldanor** [comment]
Updated the change log; yea, I think `ensure_eq` might be a bit overboard, too.

**Comment by Yamakaky** [comment]
Seems goot, thanks!

**Comment by aldanor** [comment]
@Yamakaky I was wondering if it was worth releasing a minor version with the new `with_chain()` and `ensure!()`, both of which are quite handy :)

**Comment by Yamakaky** [comment]
I can do that!

**Comment by aldanor** [comment]
❤️
